### ISSUE TITLE
Update to Timings v2

### ISSUE THREAD

**Issue by Muqsit** [pmmp/PocketMine-MP#364]
### Issue description

Timings v1 is no longer supported. So for now, /timings is useless.
[This](https://timings.aikar.co/?url=24007845) is what comes up.

### Steps to reproduce the issue

1. Start timings. (/timings on)
2. Upload timings (/timings paste)
3. Open link.

### OS and versions

* PocketMine-MP: (Doesn't matter) f2f51a8
* PHP: (Doesn't matter) 7.0.15
* Server OS: (Doesn't matter) Ubuntu 14.04
* Game version: (Doesn't matter) Win10

### Crashdump, backtrace or other files

[A Timings v1 link.
](https://timings.aikar.co/?url=24007845)

**Comment by dktapps** [comment]
Oh... that's not good.

**Comment by Muqsit** [comment]
Yeah. Figured it out just today.

**Comment by madecode15** [comment]
Timings v1 has actually been outdated for a while now, we needed to switch over anyways. Thus, timings v2 looks cleaner and gives more visual detailed reports.

**Comment by dktapps** [comment]
yeah, and who do you think has the time for upgrading?

**Comment by robske110** [comment]
I actually don't like the way timings v2 look. Maybe we could host the old code somewhere (it's open src)

**Comment by HimbeersaftLP** [comment]
And the funniest thing of all is that the page contains a link to a Spigot spoon xD

**Comment by MrPowerGamerBR** [comment]
@HimbeersaftLP fork != spoon.

I may be wrong, but isn't "spoon" just something created by the PocketMine-MP community to represent a PocketMine-MP fork that just copies and pastes code from other forks? If it is that, then Paper isn't a "spoon" because Paper is the best (and most used) Spigot fork due to it's performance improvements and API additions.

**Comment by dktapps** [comment]
Is this off-topic discussion going to end or do I have to lock the issue? As amusing as it is, it's not relevant to the topic.

**Comment by HimbeersaftLP** [comment]
Last message here (at least from me):

> @HimbeersaftLP fork != spoon.
I may be wrong, but isn't "spoon" just something created by the PocketMine-MP community to represent a PocketMine-MP fork that just copies and pastes code from other forks? If it is that, then Paper isn't a "spoon" because Paper is the best (and most used) Spigot fork due to it's performance improvements and API additions.

According to Intyre: A spoon is a "fork" which wasn't forked by just pressing the fork button, but instead uploading the code from another repo on the own one

**Comment by MrPowerGamerBR** [comment]
@HimbeersaftLP and now the last message here (at least for me):

> According to Intyre: A spoon is a "fork" which wasn't forked by just pressing the fork button, but instead uploading the code from another repo on the own one

Paper doesn't fork the original repo for (as far as I know) two reasons:
1. Spigot's repo isn't on GitHub (it is on SpigotMC Stash, so you can't fork it like you would fork, for example, PocketMine-MP's repo)
2. Due to the nature of how CraftBukkit, Spigot and Paper handle code patching, "forking" would be very redundant. (take a look at how Paper compiles, it downloads the Mojang server, patches it with CraftBukkit patches, then patches it with Spigot patches and then patches it with Paper patches)

**Comment by Muqsit** [comment]
You all are just going off topic now. Your argument isn't going to solve this issue.

I think hosting Timings v1, as @robske110 said, might be a good idea. I too hate the new look of Timings and am better off sticking to the plain simple Timings v1.

**Comment by archie426** [comment]
@Muqsit Although you are being slightly hippocritic, yes, please stop going off topic. Buuttttt @HimbeersaftLP is right about the definition of spoon (End of dicussion, now)

**Comment by dktapps** [comment]
I personally like Timings V2, but the implementation looks frightful, and there's no documentation whatsoever on upgrading or implementing it.

**Comment by Muqsit** [comment]
I might kinda agree on that. This got me excited: https://github.com/aikar/timings/issues/16

I think the project, at this point, is just trying to promote "Paper".

**Comment by MrPowerGamerBR** [comment]
@Muqsit I think that, if you ask aikar, he will explain how to implement it. (Paper isn't the only Timings v2 implementation as far as I know, Sponge and Nukkit also has a Timings v2 implementation... but of course, those two are in Java, so that's why it was easy to implement Timings v2 on them)

And don't forget to ask aikar to create a user agent for PocketMine-MP (you can also use Sponge as the user agent... but that's just a small workaround)

For a short term fix, couldn't the timings URL be changed to the one used by Spigot?

**Comment by Muqsit** [comment]
@MrPowerGamerBR Java to PHP isn't a big deal but if aikar has the time to help us, why not rather document their API usage?
Also, it isn't as easy as changing the URL :/

**Comment by xZeroMCPE** [comment]
@Muqsit @dktapps Yes, indeed. The old Timings was way better, I myself, was able to apprehend clearly.. But the V2 seems confusing...

Well, In other word, I've created a fork of the old aikar (V1) and hosted it. 

MCPETimings.com

You can start using it right-away just from *a single line of code* like this (https://github.com/TesseractTeam/Tesseract/pull/307/files)

Pardon me if this feels like i'm advertising, I just want other MCPE: Server owners to get the old timings V1 back..

**Comment by madecode15** [comment]
@zhuowei 

> The old Timings was way better

Wrong. Timingsv2 can identify problems far beyond timingsV1.

> Pardon me if this feels like i'm advertising, I just want other MCPE: Server owners to get the old timings V1 back..

You fit in the definition of advertising a crap ~fork~ ~spoon~ noob type software.

I don't like the menu on MCPE 0.15+, how about we create a new community and go back down to MCPE 0.10.5?

**Comment by Muqsit** [comment]
@madecode15 It is a temporary solution until someone figures out how the Timings v2 API works.

**Comment by flaxues** [comment]
Well mcpetimings.com is down...

**Comment by HimbeersaftLP** [comment]
I hope this doesn't count as advertising, cause I just want to help: timings.imgcl.co

**Comment by dktapps** [comment]
If you've updated your pocketmine.yml recently, you'll find an option to choose your host to use for timings. (In case anybody didn't know)

**Comment by HimbeersaftLP** [comment]
> If you've updated your pocketmine.yml recently, you'll find an option to choose your host to use for timings. (In case anybody didn't know)

That's why I posted that URL :P

**Comment by dktapps** [comment]
yeah... some people have been hardcoding it though.

**Comment by flaxues** [comment]
I'm using ImagicalMines timings for some time now. Just wanted to mention it

**Comment by dktapps** [comment]
Reducing priority on this since we now have our own timings host, so this isn't a major issue anymore.

**Comment by Muqsit** [comment]
Shall this be closed? I don't think anyone wants to shift to the new timings anyway.

**Comment by Driesboy** [comment]
nope i like the old more

**Comment by TheDiamondYT1** [comment]
@Driesboy The old one is plain and ugly, and doesn't give much information

**Comment by dktapps** [comment]
I have no intention to fill this issue for reasons discussed elsewhere (no reference implementation, no specification, little benefit to doing so as opposed to just using a normal profiler format), so I'm going to close this issue. Timings v2 has terrible support and the work needed to implement it, without any spec or reference, is not worth the benefits.

This does not mean that timings won't be improved in the future - I do have plans to improve our fork of Timings v1 and Timings in general, but I have no intention whatsoever to implement v2.

TL;DR: Timings v2 sucks.
